app/Console/Commands/GeneratePdfs.php

```diff
@@ -103,36 +84,67 @@ protected function generatePdf($model, $route = null)
 
         set_time_limit(0);
         $html = Http::get($fullUrl, ['print' => true])->body();
-
-        $pdfFileName = self::pdfFileName($model);
-        $pdfPath = storage_path('app/' . $pdfFileName);
-        $prince->convertStringToFile($html, $pdfPath);
-
-        if (config('aic.pdf_s3_enabled')) {
-            // Stream the file to S3; be sure to set `AWS_BUCKET` in `.env` and otherwise configure credentials
-            Storage::disk('pdf_s3')->putFileAs(self::STORAGE_PATH, new File($pdfPath), $pdfFileName, 'public');
+        $fileName = self::fileName($model);
+        $class = class_basename($model);
+        if ($prince->convertStringToFile($html, self::localPath($fileName))) {
+            $this->storePdf($fileName);
+        } else {
+            throw new \Exception("Prince was unable to generate a PDF for {$class} with ID {$model->id}");
         }
 
-        if ($model->pdf_download_path != self::pdfDownloadPath($pdfFileName)) {
-            $model->pdf_download_path = self::pdfDownloadPath($pdfFileName);
+        $model->pdf_download_path = self::downloadPath($fileName);
+        if ($model->isDirty('pdf_download_path')) {
```

Comment:
Would `pdf_download_path` be set to the same value every time this code runs for a given model? If so, why would the `isDirty` test come back true?